conformance/tests/gateway-invalid-route-kind.yaml

```diff
@@ -30,5 +30,5 @@ spec:
         namespaces:
           from: All
         kinds:
-        - kind: UDPRoute
+        - kind: InvalidRoute
```

Comment:
This test is about checking whether the route is compatible with the protocol type. The test description states 
```
A Gateway in the gateway-conformance-infra namespace should fail to become ready if explicitly supports any route type incompatible with the protocol type, even if there are types which are valid as well.
```
And for this reason, having a `UDPRoute` with protocol `HTTP` is considered invalid. 
While the test of an invalid Route kind is something we do not have yet as far as I see from the code. Can you confirm this was your intention @robscott? I guess we can create an issue for this missing test and implement it in another PR. WDYT?

Reply:
Good question, I'd wanted to take a step back from that as we're not prescriptive in the spec about which protocols are compatible with which route types. I think the most important takeaway here is that we want implementations to have consistent behavior if someone tries to attach an unsupported/invalid route kind to a Gateway. This slightly alters the purpose of the original test but I think it results in a clearer definition that more closely matches the spec.

Reply:
I've updated the description of this test to more clearly match what it's actually doing now, hopefully the approach makes sense. 

It would probably be good to create a follow up issue like you're suggesting, but I think that issue should be focused on which route types we consider compatible with which Gateway protocols. Until we define that, I don't think we should have conformance tests that make assumptions about which combinations are valid/invalid.

Reply:
It makes sense. I created #1515 to track it.